### ISSUE TITLE
[Mobile apps][FHIR] JSON structure > Inconsistency between iOS and android

### ISSUE THREAD

**Issue by Lathan-btc** [GoogleCloudPlatform/fda-mystudies#4987]
Steps:
1. Sign up or sign in to the mobile app (Both android and iOS)
2. Enroll to the study
3. Click on the form step questions
4. Click on the Skip button and submit the response
5. Go to the FHIR store and observe

AR: There is an inconsistency between both iOS JSON and Android JSON in FHIR 
ER: There should be a consistency between both android and iOS

**iOS JSON structure**

```
{
  "authored": "2022-09-22T14:55:29.717+05:30",
  "id": "bce06a2a-783d-4371-8a56-dede90fed2f5",
  "identifier": {
    "type": {
      "text": "questionnaire"
    },
    "use": "official",
    "value": "CopyOfImportedM@2c9080af82e94ef30182eee70f2b0016@5380317f-5659-4182-ad34-3aa4c5163a9f@q6@1"
  },
  "item": [
    {
      "definition": "scale",
      "linkId": "Step1_Scale",
      "text": "Question Scale"
    },
    {
      "definition": "continuousScale",
      "linkId": "Short2_CS",
      "text": "Question CS"
    },
    {
      "definition": "textScale",
      "linkId": "Short3_TextScal",
      "text": "Question Text scale"
    },
    {
      "definition": "valuePicker",
      "linkId": "short4_VP",
      "text": "Question VP"
    },
    {
      "definition": "textChoice",
      "linkId": "short6_TC",
      "text": "Question TC"
    },
    {
      "definition": "boolean",
      "linkId": "short_Boolean",
      "text": "Question boolean"
    },
    {
      "definition": "numeric",
      "linkId": "short_Numeric",
      "text": "Question Numeric"
    },
    {
      "definition": "timeOfDay",
      "linkId": "short_",
      "text": "time of the day"
    },
    {
      "definition": "date",
      "linkId": "short_date",
      "text": "question date"
    },
    {
      "definition": "text",
      "linkId": "text",
      "text": "text"
    },
    {
      "definition": "email",
      "linkId": "email",
      "text": "email"
    },
    {
      "definition": "timeInterval",
      "linkId": "timeinterval_sh",
      "text": "time interval"
    },
    {
      "definition": "height",
      "linkId": "height",
      "text": "height"
    },
    {
      "definition": "location",
      "linkId": "loc",
      "text": "location"
    },
    {
      "definition": "grouped",
      "item": [
        {
          "definition": "scale",
          "linkId": "formscale",
          "text": "Scale"
        },
        {
          "definition": "continuousScale",
          "linkId": "CS_STK",
          "text": "CS?"
        },
        {
          "definition": "textScale",
          "linkId": "ts_stk",
          "text": "ts"
        },
        {
          "definition": "valuePicker",
          "linkId": "vp_sk",
          "text": "vp"
        },
        {
          "definition": "textChoice",
          "linkId": "TC_SK",
          "text": "Tc"
        },
        {
          "definition": "boolean",
          "linkId": "boolean",
          "text": "boolean Question"
        },
        {
          "definition": "numeric",
          "linkId": "numeric_qsk",
          "text": "numeric"
        },
        {
          "definition": "timeOfDay",
          "linkId": "Timeoftheday_QS",
          "text": "timeoftheday_quest"
        },
        {
          "definition": "date",
          "linkId": "date",
          "text": "date"
        },
        {
          "definition": "text",
          "linkId": "text_stk",
          "text": "text"
        },
        {
          "definition": "email",
          "linkId": "email_qst",
          "text": "Email"
        },
        {
          "definition": "timeInterval",
          "linkId": "TI",
          "text": "TI"
        },
        {
          "definition": "height",
          "linkId": "Height_ST",
          "text": "Height"
        },
        {
          "definition": "location",
          "linkId": "location+STK",
          "text": "location"
        }
      ],
      "linkId": "formsteptitle",
      "text": "formsteptitle"
    }
  ],
  "meta": {
    "lastUpdated": "2022-09-22T09:24:45.559365+00:00",
    "versionId": "MTY2MzgzODY4NTU1OTM2NTAwMA"
  },
  "questionnaire": "/fhirStores/FHIR_CopyOfImportedM/fhir/Questionnaire/b9c41fe7-d7b0-4432-ac30-e83f5737a352/_history/MTY2MjExODUyNTU1OTg1MDAwMA",
  "resourceType": "QuestionnaireResponse",
  "source": {
    "reference": "Patient/a43750eb-d869-464e-a5a6-ddb7c291d7c4",
    "type": "Patient"
  },
  "status": "completed"
}
```

**Android JSON structure**

```
{
  "authored": "2022-09-22T17:11:53.775+05:30",
  "id": "bef3c00f-10f5-4906-8ad0-4aa90c78ec5a",
  "identifier": {
    "type": {
      "text": "questionnaire"
    },
    "use": "official",
    "value": "CopyOfImportedM@2c9080af82e94ef30182eee70f2b0016@f0898c65-4a4c-4556-8b2a-f1de00ae81c7@q6@1"
  },
  "item": [
    {
      "definition": "scale",
      "linkId": "Step1_Scale",
      "text": "Question Scale"
    },
    {
      "definition": "continuousScale",
      "linkId": "Short2_CS",
      "text": "Question CS"
    },
    {
      "definition": "textScale",
      "linkId": "Short3_TextScal",
      "text": "Question Text scale"
    },
    {
      "definition": "valuePicker",
      "linkId": "short4_VP",
      "text": "Question VP"
    },
    {
      "definition": "textChoice",
      "linkId": "short6_TC",
      "text": "Question TC"
    },
    {
      "definition": "boolean",
      "linkId": "short_Boolean",
      "text": "Question boolean"
    },
    {
      "definition": "numeric",
      "linkId": "short_Numeric",
      "text": "Question Numeric"
    },
    {
      "definition": "timeOfDay",
      "linkId": "short_",
      "text": "time of the day"
    },
    {
      "definition": "date",
      "linkId": "short_date",
      "text": "question date"
    },
    {
      "definition": "text",
      "linkId": "text",
      "text": "text"
    },
    {
      "definition": "email",
      "linkId": "email",
      "text": "email"
    },
    {
      "definition": "timeInterval",
      "linkId": "timeinterval_sh",
      "text": "time interval"
    },
    {
      "definition": "height",
      "linkId": "height",
      "text": "height"
    },
    {
      "definition": "location",
      "linkId": "loc",
      "text": "location"
    },
    {
      "definition": "grouped",
      "linkId": "formsteptitle",
      "text": "formsteptitle"
    }
  ],
  "meta": {
    "lastUpdated": "2022-09-22T11:42:01.373555+00:00",
    "versionId": "MTY2Mzg0NjkyMTM3MzU1NTAwMA"
  },
  "questionnaire": "/fhirStores/FHIR_CopyOfImportedM/fhir/Questionnaire/b9c41fe7-d7b0-4432-ac30-e83f5737a352/_history/MTY2MjExODUyNTU1OTg1MDAwMA",
  "resourceType": "QuestionnaireResponse",
  "source": {
    "reference": "Patient/006ec87b-6f61-4000-858c-d1a0fe508e0f",
    "type": "Patient"
  },
  "status": "completed"
}
```

**Comment by harshith2590** [comment]
This issue can be tested once IOS team has worked on Skip. @Lathan-btc @GovindPrasadBostonTechnologyCorporation  Kindly take a note of this.

**Comment by Lathan-btc** [comment]
Issue is fixed in the latest iOS dev build 3.0(180)

**Comment by nishit-v** [comment]
Issue is fixed in latest QA iOS build V3.0.189
